assets/js/frontend/blinds.js

```diff
@@ -4,33 +4,83 @@ app
   .controller("blindsCtrl", function($scope, $http, DataLoader) {
 
     $scope.selected_attribute = '';
-    $scope.productQuantity = 1;
-    $scope.showLoader = false; 
-
-    $scope.bd_get_price = function(width, drop, selectedSwatch, qty) {
-
-      if (!width || !drop || !selectedSwatch || parseInt(drop) < 500) return;
-
-      $scope.showLoader = true; 
-
-      DataLoader.getBlindsPrices(width, drop, selectedSwatch).then(function(response) {
-        console.log(response.data);
-        if (!response.data.price) return;
-        // Price without tax
-        let priceNoTax = response.data.price;
-        // Tax rate
-        let tax = response.data.tax[1].rate;
-        // Price with tax
-        let priceNoQty = (priceNoTax * tax / 100) + priceNoTax;
-        // Price x product quantity
-        let price = priceNoQty * qty;
-        // Add price to scope
-        $scope.finalPrice = price.toFixed(2);
-        // Add currency symbol to scope
-        $scope.currencySymbol = response.data.currency;
+    $scope.productQuantity = jQuery('.quantity').find('input').val();
+    $scope.showLoader = false;
+
+    $scope.bd_get_price = function(width, drop, selectedSwatch, qty, restraints) {
+
+      //console.log(restraints[selectedSwatch][0]);
+      var bd_dims = selectedSwatch ? restraints[selectedSwatch][0] : {};
+
+      if (!selectedSwatch) {
+        jQuery('.notice').html('Please Choose a finish').css('color', 'red');
+        jQuery('.single_variation_wrap, .final-price').fadeOut();
+        return;
+
+      } 
+      else if (!width || !drop) {
+        jQuery('.notice').html('Please Enter Dimensions').css('color', 'red');
+        jQuery('.single_variation_wrap, .final-price').fadeOut();
+        return;
+
+      } 
+      else if (width > parseInt(bd_dims.max_width)) {
+        jQuery('.notice').html(blinds.max_width_error + ': '+parseInt(bd_dims.max_width)+'. Please consider splitting the product into two. Or feel free to contact us for assistance.').css('color', 'red');
+        jQuery('.single_variation_wrap, .final-price').fadeOut();
+        return;
+
+      }
+      else if (width < parseInt(bd_dims.min_width)) {
+        jQuery('.notice').html(blinds.min_width_error +': '+parseInt(bd_dims.min_width)).css('color', 'red');
+        jQuery('.single_variation_wrap, .final-price').fadeOut();
+        return;
+
+      }
+      else if (drop > parseInt(bd_dims.max_drop)) {
+        jQuery('.notice').html(blinds.max_drop_error + ': '+parseInt(bd_dims.max_drop)).css('color', 'red');
+        jQuery('.single_variation_wrap, .final-price').fadeOut();
+        return;
+
+      } 
+      else if (drop < parseInt(bd_dims.min_drop)) {
+        jQuery('.notice').html(blinds.min_drop_error + ': '+parseInt(bd_dims.min_drop)).css('color', 'red');
+        jQuery('.single_variation_wrap, .final-price').fadeOut();
+        return;
+
```

Comment:
Need to sort this jQuery out - replace with angular.element or move to directive